torch/onnx/_internal/exporter/_decomp.py

```diff
@@ -83,18 +81,14 @@ def create_onnx_friendly_decomposition_table(
         Dict[torch._ops.OperatorBase, Callable]: A dictionary that maps op overloads to their corresponding
         decomposition functions.
     """
-    decomposition_table: dict[torch._ops.OperatorBase, Callable] = {}
-
-    # NOTE: If we import torch._decomp, we will get RuntimeError: Only a single
-    # TORCH_LIBRARY can be used to register the namespace nvprims; please put all of your
-    # definitions in a single TORCH_LIBRARY block.
-    for op_overload, decomp_fn in torch._decomp.decomposition_table.items():  # type: ignore[attr-defined]
-        # Skip decomposition for op_overload as long as that op_overload has a corresponding ONNX
-        # symbolic function.
-        # NOTE: Do not skip torch._refs decomps. They are fine because otherwise the model is
-        # not exportable anyways.
-        if op_overload in onnx_registered_ops:
-            continue
-        decomposition_table[op_overload] = decomp_fn
+    decomposition_table: dict[torch._ops.OperatorBase, Callable] = (
+        torch._decomp.core_aten_decompositions()
+    )
+
+    preserve_ops = get_preserve_ops()
+
+    for op_overload in list(decomposition_table.keys()):
```

Comment:
cc @justinchuby

Reply:
Thanks: Just making sure: Supposed we have e.g. aten.stft in onnx_registered_ops, but since they were by default decomposed and not in preserve_ops, we want them to continue to be decomposed. (context, our stft was not implemented correctly but was registered nevertheless. Since it was decomposed earlier no error was caught. If it is now retained then we would start seeing errors)

Is that possible? Assuming you have a table of the CompositeImplicitAutograd ops I think I would be possible to update the table with this complete list? If not that's ok too: I would still appreciate a list of these ops so I can clean up our faulty implementation.

Reply:
I rewrote the logic. In this case, aten.stft will still be decomposed. After this change, you can think of core aten having two seperate tables combined together: (1) CIA decomps + (2) Python decomps. Previously only (2) was part of the core aten decomp. Here is how i rewrote the ONNX decomp:

1) I first collect all CIA decomps (delta between old and new behaviour)
2) I delete all CIA decomps that are in get_preserve_ops list from onnx 

At this point, we have a table that has CIA decomps except for the ones that are specified by ONNX. 

3) We walk the torch._decomp.decompositions which contains the prim decomps and don't add the onnx_registered_ops to this list. 

4) We combine (2) and (3) into one table. 

By this logic, aten.stft, a CIA op, will automatically be decomposed as it doesn't show up in the get_preserve_ops list. When we add entries in (3), we iterate over torch._decomp.decompositions instead of core_aten_decomp, so aten.stft won't be deleted from the table we are passing into export.

Reply:
Thank you! That sounds perfect